File: client/src/models/Rider.ts
import { z } from "zod";

export const riderSchema = z.object({
  firstname: z.string(),
  lastname: z.string(),
  initials: z.string(),
  country: z.string(),
  riderId: z.number(),
});
export type Rider = z.infer<typeof riderSchema>;

File: api/Infrastructure/Database/TableModels/Rider.cs
namespace SpoRE.Infrastructure.Database;

public class Rider
{
    public int RiderId { get; set; }

    public string PcsId { get; set; }

    public string Country { get; set; }

    public string Firstname { get; set; }

    public string Lastname { get; set; }

    public string Initials { get; set; }

    public virtual ICollection<RiderParticipation> RiderParticipations { get; } = [];
}
